### PR TITLE
updated nested field logic

### PR DESCRIPTION
**Pull request by darshan-iterable**
## JIRA Ticket(s) if any

* [MOB-9308](https://iterable.atlassian.net/browse/MOB-9308)

## Description
Updated logic same as Android [PR comments](https://github.com/Iterable/iterable-android-sdk/pull/778#discussion_r1737260636) to maintain consistency in functionality

[MOB-9308]: https://iterable.atlassian.net/browse/MOB-9308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/anonymousUserTracking/criteriaCompletionChecker.ts [hunk 1]
```diff
@@ -351,21 +351,14 @@ class CriteriaCompletionChecker {
 
       if (field.includes('.')) {
         const fields = field.split('.');
-        if (Array.isArray(eventData[fields[0]])) {
-          return eventData[fields[0]]?.every((item: any) => {
+        const firtElement = eventData?.[fields[0]];
```
Comment (mprew97) [inline src/anonymousUserTracking/criteriaCompletionChecker.ts:354]: typo here 
```suggestion
        const firstElement = eventData?.[fields[0]];
```

**Review by mprew97 — APPROVED** [review]
One nit but otherwise lgtm

File: src/anonymousUserTracking/criteriaCompletionChecker.ts (hunk 1 above)
Comment (darshan-iterable) [reply src/anonymousUserTracking/criteriaCompletionChecker.ts]: Fixed typo

**Comment by davidtruong** [comment]
Verified logic for nested works properly

**Pull request merged**